File: easy/binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def dfsinorder(tree, resultant):
            if tree != None:
                dfsinorder(tree.left, resultant)
                resultant.append(tree.val)
                dfsinorder(tree.right, resultant)
        result = []
        dfsinorder(root, result)
        return result

File: easy/binary-tree-inorder-traversal/binary-tree-inorder-traversal.cpp
class Solution {
public:
    vector<int> res;
    void dfs(TreeNode* node) {
        if (node) {
            dfs(node->left);
            res.push_back(node->val);
            dfs(node->right);
        }
    }
    vector<int> inorderTraversal(TreeNode* root) {
        dfs(root);
        return res;
    }
};
